app/controllers/coronavirus_form/check_answers_controller.rb

```diff
@@ -67,15 +69,17 @@ def items
   end
 
   def additional_product_index
-    items.index { |item| item[:field] === t("coronavirus_form.questions.additional_product.title") }
+    @additional_product_index ||= items.index do |item|
+      item[:field] == t("coronavirus_form.questions.additional_product.title")
+    end
   end
 
   def items_part_1
-    items.select.with_index { |_, index| index < additional_product_index }
+    items.slice(0, additional_product_index)
   end
 
   def items_part_2
-    items.select.with_index { |_, index| index > additional_product_index }
+    items.slice(additional_product_index + 1..)
```

Comment:
ditto here, really nice. So the 1.. == from the 2nd value to the end of the range right?

Reply:
Sorry, it's not super clear. There's probably a better way to do this!

We have an element at a given index, and we want to puts the elements either side of that index into two arrays. For example, in an array 1-10, we want to put 1,2,3,4,5,6 into one list, and 8,9,10 into another, and exclude 7 from both:

```ruby
items = (1..10).to_a   # [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
index = 6              # 7
items.slice(0, index)  # [1, 2, 3, 4, 5, 6]
items.slice(index+1..) # [8, 9, 10]
```

Reply:
ah gotcha, no that makes sense, just not a pattern i'd seen before!
it's cool.